server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -285,6 +285,18 @@ public class IndicesService extends AbstractLifecycleComponent
         Property.Dynamic
     );
 
+    /**
+     * This setting is used to restrict creation or updation of index where the `index.translog.durability` index setting
+     * is set as ASYNC if enabled. If disabled, any of the durability mode can be used and switched at any later time from
+     * one to another.
+     */
+    public static final Setting<Boolean> CLUSTER_REMOTE_INDEX_RESTRICT_ASYNC_DURABILITY_SETTING = Setting.boolSetting(
```

Comment:
should we do something like `"cluster.remote_store.index.restrict.durability": "async"` ,  instead of a boolean. 

This would allow us to restrict sync durability as well in future if required.

Reply:
@linuxpi This is very specific to restricting the async mode since it compromises on durability on account of fire and forget nature. In future, we might want to only have sync mode for remote enabled indexes.

Reply:
Is this setting specific to remote store backed indexes?

Reply:
Yes, correct.

Reply:
Why is it specific to remote indexes?

Reply:
The async durability compromises the request level durability guarantees offered by remote store due to it's fire and forget nature. There can be cases where a user wants the cluster to be highly durable (where each doc is persisted durably before the ack). This setting ensures that the user does not create an index with async durability unintentionally. For document replication (& segrep), we do not want to disturb the status quo.

Reply:
Resolving the conversation @sachinpkale. Please reopen if you have any other question/concerns.